src/MongoDB.Driver/AggregateUnionWithOptions.cs

```diff
@@ -0,0 +1,48 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using MongoDB.Bson.Serialization;
+
+namespace MongoDB.Driver
+{
+    /// <summary>
+    /// Options for the aggregate $unionWith stage.
+    /// </summary>
+    /// <typeparam name="TForeignDocument">The type of the foreign document.</typeparam>
+    /// <typeparam name="TResult">The type of the result.</typeparam>
+    public sealed class AggregateUnionWithOptions<TForeignDocument, TResult>
+    {
+        private IBsonSerializer<TForeignDocument> _foreignSerializer;
+        private IBsonSerializer<TResult> _resultSerializer;
```

Comment:
This class probably isn't needed.

The withSerializer is determined by the withCollection.

The outputSerializer must be the same as the outputSerializer of the pipeline so far.